src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -4731,6 +4731,9 @@ void CodeGen::genPushCalleeSavedRegisters()
     }
 #endif // DEBUG
 
+    // Sign LR as part of Pointer Authentication (PAC) support
+    GetEmitter()->emitPacInProlog();
```

Comment:
Either in this PR or in a follow-up PR, you might want to consider moving the pac instructions closer to where we save `lr` on the stack instead of doing it at the top all the time. That way, you could also have a way to get the `sp` address where the `lr` was saved and then use that address to build up the original `sp` used to strip/auth the `lr` in runtime layer.

![image](https://github.com/user-attachments/assets/e11e3861-5ced-4f16-98a2-62a4057669a2)

![image](https://github.com/user-attachments/assets/820bc463-eeac-4ddd-851c-aceb0d6cd323)

Reply:
Yes, number of things will need to be redone to get SP signing in place.